sources/api/apiserver/src/server/ephemeral_storage.rs

```diff
@@ -92,7 +92,13 @@ pub fn initialize(
     info!("initializing ephemeral storage disks={:?}", disks);
     // with a single disk, there is no need to create the array
     let device_name = match disks.len() {
-        1 => disks.first().expect("non-empty").clone(),
+        1 => {
+            let disk_name = disks.first().expect("non-empty").clone();
+            let link_name = format!("{RAID_DEVICE_DIR}{RAID_DEVICE_NAME}");
+            std::os::unix::fs::symlink(disk_name, &link_name)
+                .context(error::DiskSymlinkFailureSnafu)?;
+            link_name
```

Comment:
Here you're fixing a pre-existing bug - we effectively lost track of which disk we used during the "init" operation. That didn't matter in the case where we use all ephemeral disks, but could have mattered on an instance that had three (`nvme1`, `nvme2`, `nvme3`) where the "init" request specified only one (`nvme2`) so we didn't end up creating an array.

The symlink approach is a nice, cheap way to preserve the state we need, but it can cause trouble to create links in "owned" directories under `/dev`, like `/dev/md` (which `mdadm` "owns").

Instead I'd suggest using [pathdiff::diff_paths](https://docs.rs/pathdiff/latest/pathdiff/fn.diff_paths.html) and creating a relative symlink from `/dev/disk/ephemeral-storage` to either the single disk or EBS volume, or to the RAID-0 array.